sdcm/provision/network_configuration.py

```diff
@@ -0,0 +1,125 @@
+from dataclasses import dataclass
+
+
+class NetworkInterfaceNotFound(Exception):
+    pass
+
+
+@dataclass
+class NetworkInterface:  # pylint: disable=too-many-instance-attributes
+    ipv4_public_address: str
+    # AWS allows to have a few public IPv6 addresses per interface.
+    ipv6_public_addresses: [str]
+    # AWS allows to have a few private IPv4 addresses per interface.
+    ipv4_private_addresses: [str]
+    ipv6_private_address: str
+    dns_private_name: str
+    dns_public_name: str
+    device_index: int
+
+
+class ScyllaNetworkConfiguration:
+    LISTEN_ALL = '0.0.0.0'
+
+    def __init__(self, network_interfaces, scylla_network_config):
```

Comment:
Not sure I understand what do you mean

Reply:
using dataclass or pydantic is nice to have, but we don't need to change it now

Reply:
yes, we don't need - it's not a bug per se.
When adding new code I was hoping to have it with better design - like possibility to use classes attributes (with help of type hints) instead of finding all the possible values in dict and then choosing one. Also having things as a classes, we can validate upon class creation/modification instead of separating it. It doesn't have to be dataclass.